### DISCUSSION TITLE
How can I retrieve a list of all teams?

### DISCUSSION THREAD

**Discussion by ghost** [uhd-urz/elAPI#124] [Q&A] [answered]
Hi. 
I am interested in a list of all teams in our elabftw-instance. 
How can I use elapi for that? What would the snippet look like? 

Thank you!

**Comment by mhxion** [comment, marked as answer]
Hi @LukasCBossert. If you want to do it via the command line, you can get a list of teams with just `elapi get teams`, or export it to a file with `elapi get teams --format csv --export`. If you want to do it with Python, then the following:

```py
from elapi.api import GETRequest

session = GETRequest()  # Define a HTTP session
teams = session("teams", endpoint_id=None)  # First argument is the endpoint name. This line is when the GET request is actually sent.
# You don't necessarily have to pass endpoint_id=None, that's already the default.
# Replace None with a team ID, if you want information about that team only.
print(teams.json())  # Prints list of all teams as a list of Python dictionaries (each dictionary contains one team)
```
Hope that helps.

The following is not longer necessary since elAPI version `2.1.0`. See #129.

~~There's one small (but can be cruicial) thing that I forgot/missed that @alexander-haller just found out. elAPI intends to give you more control when you're using it in a Python script. Which is why it doesn't do some "non-essential validations", like checking if your API key (or token) is actually valid among many other things. If you want elAPI to run those validations you need to add two more boilerplate lines of code:~~

~~from elapi.api import GETRequest~~
~~from elapi.configuration import reinitiate_config~~

~~reinitiate_config()~~
~~# <Everything else is the same as before>~~


~~elAPI does these validations automatically when you run it via the CLI or when you use your script as an elAPI plugin so you'd not have to include this `reinitiate_config()` once you feel like you script has sufficiently matured and can be used as a plugin.~~

**Reply by ghost** [reply]
Thank you very much @mhxion I’ll give it a try.
